libraries/cms/application/cms.php

```diff
@@ -114,103 +114,15 @@ public function __construct(JInput $input = null, Registry $config = null, JAppl
 		}
 
 		// Enable sessions by default.
-		if (is_null($this->config->get('session')))
+		if (is_null($this->getContainer()->get('config')->get('session')))
```

Comment:
Having a container as root object which manages the application, configs, inputs and so on would require duplicated code in the index.php for the site and the administrator index.php file, as example. It would increase the complexity to create an application wherever it is used.
From my understanding the component will always require an application context it is running in. To start an app like

    $app = new JApplicationAdministrator();
    $app->execute();

invites me more to create my own apps than this example https://gist.github.com/mbabker/a9b24a472c53f580580e. I really don't see the benefit to fill the container in the index.php file than to let the application manage it by its own.